### PR TITLE
Circuit composition wraps per default

### LINKED ISSUE #6147: compose should not unroll (by default) when using the circuit library
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

The idea of having a circuit library is to build up larger circuits and have simple visualizations that we can recognize when using them. I am fine having a non default option to unroll circuit when you compose it but by default the combined circuit should look like building blocks of the libraries. 

As an example see this figure 
Screen Shot 2021-04-03 at 11 07 27 AM

It should make output (by default)

Screen Shot 2021-04-03 at 11 08 06 AM

@ajavadia I know we talked about this but just making issue so we remember.

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Add a ``wrap`` keyword argument to ``QuantumCircuit.compose`` to set whether the composed circuit should be wrapped in an instruction or not.

 Per default, wrap the other circuit, to allow a better overview of the individual circuit blocks (especially if attaching library circuits) and be closer to the behavior of ``append`` so that in long-term we can remove that one and keep only ``compose``. Note that this changes how the default of how `circuit.compose(other)` looks like, but it's no functional change.

Closes #6147 (but does a bit more, see also the discussion in that issue).

### Details and comments

* `compose` get's a new `wrap` argument that's `True` by default
* `append` also get's a `front` argument to append infront of a circuit to streamline the usage with compose (but mainly to simplify the logic in the code)
* the library circuits `QuantumVolume`, `Permute` and `IQP` don't return a circuit with a single wrapped instruction, since compose now wraps and to improve consistency with the rest of the library

### CODE REVIEW

File: qiskit/circuit/quantumcircuit.py [hunk 1]
```diff
@@ -643,7 +643,7 @@ def extend(self, rhs):
 
         return self
 
-    def compose(self, other, qubits=None, clbits=None, front=False, inplace=False):
+    def compose(self, other, qubits=None, clbits=None, wrap=True, front=False, inplace=False):
```
Comment (Cryoris) [inline qiskit/circuit/quantumcircuit.py:646]: @jaygambetta (I need a message to send you a link to the code)

**Comment by jaygambetta** [comment]
thanks looking forward to seeing this but if we dont wrap by default this does not close the issue we need a second pr for the libraries.

**Comment by Cryoris** [comment]
yes, I'll refactor this one and open a new one for wrapping the library circuits 👍🏻

**Comment by Cryoris** [comment]
Closing in favor of #6634 and #6659.

**Pull request closed without merging**